src/Og.php

```diff
@@ -196,17 +194,17 @@ public static function getEntityGroups(EntityInterface $entity, array $states =
 
     /** @var \Drupal\og\Entity\OgMembership $membership */
     foreach ($memberships as $membership) {
-      static::$entityGroupCache[$identifier][$membership->getGroupEntityType()][$membership->id()] = $membership->getGroup();
+      static::$cache[$identifier][$membership->getGroupEntityType()][$membership->id()] = $membership->getGroup();
```

Comment:
It also seems wasteful that we are actually loading the full OgMembership objects, but then we only return the ID. This means if we need the full objects on the calling side that we will have to load them again.

Wouldn't it be more logical to simply return the full OgMembership objects in this method? We don't need to preload the groups then in an inefficient loop, and it seems cleaner overall - returning two types in OO code is often seen as a "code smell". We could rename it to `getUserMemberships()`.

Then again, if you query the memberships, it's probably because you are interested more in the groups than in the memberships themselves.